python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -215,6 +220,52 @@ def execute(self, settings: CLISettings) -> None:
                             fetched_data,
                         )
 
+    def validate_hashes_signal_type(
+        self,
+        hashes_grouped_by_prefix: t.Dict[t.Optional[t.Type[SignalType]], t.Set[str]],
+        signal_types: t.List[t.Type[SignalType]],
+    ) -> None:
+        if (
+            len(hashes_grouped_by_prefix) > 2
+            and None in hashes_grouped_by_prefix.keys()
+        ):
+            raise CommandError(
+                f"Error: Provided more than one SignalType and some hashes are missing a prefix",
+                2,
+            )
+        if self.only_signal:
+            if (
+                self.only_signal not in hashes_grouped_by_prefix.keys()
```

Comment:
nit: You query is on a dict, which by default `in` looks at keys with O(1), this .keys() will create an iterator that takes O(n) to evaluate